src/SW_Flow_lib.c

```diff
@@ -3277,6 +3277,21 @@ void soil_temperature(
     for (i = 0; i < nlyrs; i++) {
         maxTempSoil[i] = meanTempSoil[i] + (temperatureRange[i] / 2);
         minTempSoil[i] = meanTempSoil[i] - (temperatureRange[i] / 2);
+
+        if (minTempSoil[i] < -100 || maxTempSoil[i] > 100) {

```

Comment:
I suggest we use `LT()` and `GT()` in instances all instances except for the ones checking for a positive value, since this does not rely on floating point precision.